src/main/java/com/google/sps/data/OrganizationUpdater.java

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Arrays;
+import com.google.appengine.api.datastore.Entity;
+import com.google.sps.data.GivrUser;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+
+public final class OrganizationUpdater {
+
+  private Entity entity;
+
+  public OrganizationUpdater(Entity entity) {
+    this.entity = entity;
+  }
+
+  public Entity getEntity() {
+    return this.entity;
+  }
+
+  public void updateOrganization(HttpServletRequest request, GivrUser user, boolean forRegistration) throws IllegalArgumentException{
+    Set<String> requiresMaintainer = new HashSet<String>();
+    Set<String> requiresModerator = new HashSet<String>();
+    Map<String, String> properties = new HashMap<String,String>();
+    boolean isMaintainer = user.isMaintainer();
+    boolean isModerator = user.isModerator();
+
+    requiresMaintainer.add("isApproved");
+    requiresModerator.add("moderatorList");
+
+    // Format is Form Name, Entity Property 
+    properties.put("org-name", "orgName");
+    properties.put("org-email", "orgEmail");
+    properties.put("org-street-address", "orgStreetAddress");
+    properties.put("org-phone-num", "orgPhoneNum");
+    properties.put("org-url", "orgUrl");
+    properties.put("org-description","orgDescription");
+    properties.put("approval", "isApproved");
+    properties.put("moderator-list", "moderatorList");
+
+    for(Map.Entry<String, String> entry : properties.entrySet()) {
+      String propertyKey = entry.getValue();
+
+      // if updating for registering an organization then do not want to consider fields that require maintainer or moderator permissions
+      if(forRegistration && (requiresModerator.contains(propertyKey) || requiresMaintainer.contains(propertyKey))) {
```

Comment:
this will be a bit simpler to read if you used some helper vars;


bool propertyRequiresMaintainer = requiresMaintainer.contains(propertyKey);
bool propertyRequiresModerator = requiresModerator.contains(propertyKey);

Then the rest of the code will read a bit easier (and you won't recompute the set membership)

Reply:
I would even add an additional:

bool propertyRequiresAuth = propertyRequiresMaintainer || propertyRequiresModerator;

now this condition becomes

if (forRegistration && propertyRequiresAuth) {
  continue;
}

which really makes it easy to read